### ISSUE TITLE
Swift 4 generated mock causes an error - invalid conversion...

### ISSUE THREAD

**Issue by markturner** [Brightify/Cuckoo#151]
Hi guys,

So I've just got Xcode 9 and have followed the instructions to convert to swift 4. I'm using 0.10.0 of Cuckoo. Everything is fine except GeneratedMocks.swift has this code in it:

```
    // ["name": "httpClient", "accesibility": "", "@type": "InstanceVariable", "type": "HttpClientProtocol", "isReadOnly": false]
     override var httpClient: HttpClientProtocol {
        get {
            return cuckoo_manager.getter("httpClient", original: observed.map { o in return { () -> HttpClientProtocol in o.httpClient }})
        }
        
        set {
            cuckoo_manager.setter("httpClient", value: newValue, original: observed != nil ? { self.observed?.httpClient = $0 } : nil)
        }
        
    }
```

Which causes the following error when building:

```
Invalid conversion from throwing function of type '((Schedule) throws -> _) throws -> _?' to non-throwing function type '((Schedule) throws -> () -> HttpClientProtocol) -> ((()) -> _)?'
```

Do I need to change my class to prevent this? Or anything else I can do? I don't really get much from that error to be honest! This has worked today but I can't get past this error now, I haven't knowingly changed anything else.

Any advice you can give would be appreciated, let me know if you need any further info.

Many thanks!

**Comment by TadeasKriz** [comment]
Fixed in `0.10.1`.

**Comment by arkilis** [comment]
Still exists in 0.10.2
